### PR TITLE
Live outbound references for map.

### PR DESCRIPTION
**Pull request by polina-c**
RELEASE_NOTE_EXCEPTION=[behind flag]

Screenshot 2023-02-06 at 10 50 04 AM

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart [hunk 1]
```diff
@@ -344,8 +365,25 @@ List<DartObjectNode> createVariablesForAssociations(
     false,
     (p, e) => p || isPrimativeInstanceKind(e.key.kind),
   );
-  for (var i = 0; i < associations.length; i++) {
+  for (final i in Iterable.generate(associations.length)) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart:368]: what's the reason for this change? This requires us to create another iterable instead of just looping through the one we already have `associations`

File: packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart [hunk 2]
```diff
@@ -344,8 +365,25 @@ List<DartObjectNode> createVariablesForAssociations(
...
     final association = associations[i];
+
+    if (asReferences) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart:371]: will we eventually have to differentiate between whether this should be added as a live reference or a static reference?

File: packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart:368]: reverted

File: packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart:371]: Static references never show up here. I plan to reorganize code to make it more supportable.

**Review by kenzieschmoll — APPROVED** [review]
